articles/huffman-coding-in-python/index.md

```diff
@@ -0,0 +1,184 @@
+            elif(huffman_code[num-i-1] != ''):
+                huffman_code[num-i-2] = '0'
+                huffman_code[-1] = ['1' + symbol for symbol in huffman_code[-1]]
+            elif(huffman_code[num-i-2] != ''):
+                huffman_code[num-i-1] = '1'
+                huffman_code[-2] = ['0' + symbol for symbol in huffman_code[-2]]
+            else:
+                huffman_code[num-i-1] = '1'
+                huffman_code[num-i-2] = '0'
+
+            position = self.position(val, i)
+            probability = self.probability[0:(len(self.probability) - 2)]
+            probability.insert(position, val)
+            if(isinstance(huffman_code[num-i-2], list) and isinstance(huffman_code[num-i-1], list)):
+                complete_code = huffman_code[num-i-1] + huffman_code[num-i-2]
+            elif(isinstance(huffman_code[num-i-2], list)):
+                complete_code = huffman_code[num-i-2] + [huffman_code[num-i-1]]
+            elif(isinstance(huffman_code[num-i-1], list)):
+                complete_code = huffman_code[num-i-1] + [huffman_code[num-i-2]]
+            else:
+                complete_code = [huffman_code[num-i-2], huffman_code[num-i-1]]
+            
+            huffman_code = huffman_code[0:(len(huffman_code)-2)]
+            huffman_code.insert(position, complete_code)
+
+        huffman_code[0] = ['0' + symbol for symbol in huffman_code[0]]
+        huffman_code[1] = ['1' + symbol for symbol in huffman_code[1]]
+        
+        if(len(huffman_code[1]) == 0):
+            huffman_code[1] = '1'
+        
+        count = 0
+        final_code = ['']*num
+
+        for i in range(2):
+            for j in range(len(huffman_code[i])):
+                final_code[count] = huffman_code[i][j]
+                count += 1
+
+        final_code = sorted(final_code, key=len)
+        return final_code
+
+string = input("Enter the string to compute Huffman Code: ")
+
+freq = {}
+for c in string:
+    if c in freq:
+        freq[c] += 1
+    else:
+        freq[c] = 1
+
+freq = sorted(freq.items(), key=lambda x: x[1], reverse=True)
+length = len(string)
+
+probabilities = [float("{:.2f}".format(frequency[1]/length)) for frequency in freq]
+probabilities = sorted(probabilities, reverse=True)
+
+huffmanClassObject = HuffmanCode(probabilities)
+P = probabilities
+
+huffman_code = huffmanClassObject.compute_code()
+
+print(' Char | Huffman code ')
+print('----------------------')
+
+for id,char in enumerate(freq):
+    if huffman_code[id]=='':
+        print(' %-4r |%12s' % (char[0], 1))
+        continue
+    print(' %-4r |%12s' % (char[0], huffman_code[id]))
+
+huffmanClassObject.characteristics_huffman_code(huffman_code)
+```
+
+#### Auxiliary functions
+We have a couple of auxiliary functions such as find_position and characteristics_huffman_code. find_position is used to insert bits to the existing code computed in the n-3 previous iterations, where n is the length.
```

Comment:
such as find_position and characteristics_huffman_code. find_position is  -> such as find_position and `characteristics_huffman_code`. `find_position` is